prez/renderers/json_renderer.py

```diff
@@ -0,0 +1,225 @@
+from itertools import chain
+
+from rdflib import Graph, URIRef, RDF, SH, Literal
+from rdflib.term import Node
+
+from prez.cache import profiles_graph_cache
+from prez.reference_data.prez_ns import ALTREXT
+
+
+class NotFoundError(Exception):
+    ...
+
+
+def _get_resource_iri(graph: Graph, profile_graph: Graph, profile: URIRef) -> Node:
+    target_classes = profile_graph.objects(profile, ALTREXT.constrainsClass)
+    for target_class in target_classes:
+        iri = graph.value(predicate=RDF.type, object=target_class)
+        if iri is not None:
+            return iri
+
+    raise NotFoundError(
+        f"No resource IRI found based on the constrained classes defined in {profile}."
+    )
+
+
+def _get_child_to_focus_predicates(
```

Comment:
Is the plan to replace the existing functions?

Reply:
Using existing function now.